lms/djangoapps/course_home_api/tasks.py

```diff
@@ -0,0 +1,64 @@
+"""
+Celery tasks used by the `course_home_api` app.
+"""
+import logging
+
+from celery import shared_task
+from django.contrib.auth import get_user_model
+from edx_django_utils.monitoring import set_code_owner_attribute
+from eventtracking import tracker
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.student.models_api import get_course_enrollment
+from lms.djangoapps.courseware.courses import get_course_blocks_completion_summary
+
+User = get_user_model()
+COURSE_COMPLETION_FOR_USER_EVENT_NAME = "edx.bi.user.course-progress"
+
+log = logging.getLogger(__name__)
+
+
+@shared_task
+@set_code_owner_attribute
+def calculate_course_progress_for_user_in_course(course_id: str, user_id: str) -> None:
```

Comment:
it's weird that this, which is only triggered by the progress tab, repeats the same calculation. does the calculation happen in the MFE and not the backend,  is that why?

I feel like ideally this would happen on the back end and be sent both to the MFE and to celery. I understand that's a lot more complex and may be outside the scope of what we're doing here, but it does fundamentally seem like a big flag to do this twice.

Reply:
Yeah, the calculation is done on the front end. The front only gets back the summary dict (with the complete/incomplete/locked unit counts).

I agree that it is a bit odd, but thought that refactoring was a bit more complex and had more moving parts, so I didn't consider doing it at this time.

Reply:
Also wanted to add, I verified that there is already an API endpoint for the progress stats, that's how the MFE gets the rough numbers it uses to calculate the percentages.